### PR TITLE
Release v8 docs

### LINKED ISSUE #1054: Readme improvement
Hello, I think that step 2 contains a lot of code snippets, thus I had some troubles finding step 3, a lot of scroll... and this made me loose some time understanding how to execute some code through the configured policy. Sorry for this user issue.

Maybe step 2 can present only one code snippet and the other policy names/behaviors/samples could be listed/linked after code snippet... after main steps in the Readme?

After reading again, the samples are a bit more detailed than table at step 1 (Retry and RetryForever vs only Retry) and they are linked from the table as quick start. I fully agree to keep all quick starts in the Readme itself.

Hoping I'm clear, best regards,

Step 2: https://github.com/App-vNext/Polly#step-2--specify-how-the-policy-should-handle-those-faults
Step 3: https://github.com/App-vNext/Polly#step-3--execute-code-through-the-policy

### LINKED ISSUE #1091: v8 API documentation
As part of the release of Polly v8 we should ensure that either before or soon after release we have reviewed all of the existing documentation to ensure that we have:

- Updated the README and wiki to cover the v8 API
- Add any relevant new documentation that isn't covered by the above (e.g. brand new features)
- Have a migration guide for moving from v7 to v8
- Have a guide on how to wrap the v7 API with with the v8 API (i.e. a partial migration)

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Closes #1091 
Closes #1054

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Comment by martincostello** [comment]
@martintmk This just needs the merge conflicts resolving and then I think this is ready to merge and then we're _done_, right?

**Comment by martintmk** [comment]
> @martintmk This just needs the merge conflicts resolving and then I think this is ready to merge and then we're _done_, right?

Do you think we should merge this even before v8.0.0 is released?

**Comment by martintmk** [comment]
> Yes because the README is embedded in the NuGet packages and it should be the v8 one, not the v7 one.
> 
> Also there's going to be not much lag between merging this and shipping (hopefully), plus it's nicer (IMHO) for the Git tag for 8.0.0 to contain the matching documentation.

Ok, making this ready for review. v7 users will still have not pointing to older documentation on the top.

File: README.md [hunk 1]
```diff
@@ -48,1025 +26,414 @@ We are a member of the [.NET Foundation](https://www.dotnetfoundation.org/about)
...
-`PolicyRegistry` has a range of further dictionary-like semantics such as `.ContainsKey(...)`, `.TryGet<TPolicy>(...)`, `.Count`, `.Clear()`, and `Remove(...)`.
-
-Available from v5.2.0.  For more detail see: [PolicyRegistry](https://github.com/App-vNext/Polly/wiki/PolicyRegistry) on wiki.
+Rate limiter strategy throws `RateLimiterRejectedException` if execution is rejected. For more details, visit the [rate limiter strategy](https://www.pollydocs.org/strategies/rate-limiter) documentation.
 
-## Asynchronous Support
+## Next steps
 
-Polly fully supports asynchronous executions, using the asynchronous methods:
+To learn more about Polly, visit the [pollydocs.org][polly-docs].
```
Comment (martincostello) [inline README.md:427]:
```suggestion
To learn more about Polly, visit [pollydocs.org][polly-docs].
```

**Review by martincostello — APPROVED** [review]
Just spotted this one type.

**Pull request merged**